src/Printers/Languages/EplPrinterCommandSet.ts

```diff
@@ -0,0 +1,489 @@
+import { WebZlpError } from '../../WebZlpError';
+import {
+    MediaPrintMode,
+    PrinterCommandLanguage,
+    PrintOrientation,
+    PrintSpeedSettings,
+    ThermalPrintMode
+} from '../Configuration/PrinterOptions';
+import { PrinterOptions } from '../Configuration/PrinterOptions';
+import { PrinterModel, PrinterModelDb } from '../Models/PrinterModel';
+import { DocumentValidationError, PrinterCommandSet } from './PrinterCommandSet';
+import * as Commands from '../../Documents/Commands';
+import { match, P } from 'ts-pattern';
+import { NumericRange } from '../../NumericRange';
+import { PrinterCommunicationOptions } from '../Communication/PrinterCommunication';
+
+/** Command set for communicating with an EPL II printer. */
+export class EplPrinterCommandSet extends PrinterCommandSet {
+    private encoder = new TextEncoder();
+
+    get commandLanugage(): PrinterCommandLanguage {
+        return PrinterCommandLanguage.epl;
+    }
+
+    get documentStartCommand(): Uint8Array {
+        // Start of any EPL document should include a clear image buffer to prevent
+        // previous commands from affecting the document.
+        return this.encodeCommand('\r\nN\r\n');
+    }
+
+    get documentEndCommand(): Uint8Array {
+        // There's no formal command for the end of an EPL doc, but just in case
+        // add a newline.
+        return this.encodeCommand('');
+    }
+
+    encodeCommand(str: string, withNewline = true): Uint8Array {
+        // Every command in EPL ends with a newline.
+        return this.encoder.encode(str + (withNewline ? '\r\n' : ''));
+    }
+
+    transpileCommand(cmd: Commands.IPrinterCommand, outDoc: Commands.CompiledDocument): Uint8Array {
```

Comment:
Made better in #20, that object is the bag of state that can be modified by a handful of commands and intentionally supports downstream effects. Instead of being the output _compiled document_ this is now a bag-o-state object that doesn't leave the command set class.

Reply:
And #22 had some follow-up work that made this further self-contained and managed.